packages/rx_bloc_cli/doc/golden_tests.md

```diff
@@ -0,0 +1,47 @@
+### Smart Widgets dependencies for Golden Tests
+
+In golden tests we want to mock the BLoCs to be able to test all UI states.Using reusable smart widgets that have their own dependency injection will make golden tests to fail. To fix this issue we need to build the smart widgets without injecting the required components during the test. This is why we added `isInTestMode` in `app_constants.dart`.
+
+To use it in custom widgets with their own dependency injection you can follow this example:
+```
+import '../../base/app/config/app_constants.dart';
+
+...
+
+class AppTodoListBulkEditPopupMenuButtonWithDependencies {
+    @override
+    Widget build(BuildContext context) {
+        final current = TodoManagementPage();
+        
+        if(isInTestMode) {
+            return current;
+        }
+        
+        return MultiProvider(
+            providers: [
+                ..._services,
+                ..._blocs,
+            ],
+            child: current,
+        );
+    }
+}
+```
+IMPORTANT: you should import `isInTestMode` from [app_constants.dart]!
```

Comment:
You can use specific markdown formatting to indicate[ different levels of alerts](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts).

Reply:
Done